### ISSUE TITLE
Default type imported with `import type` not flagged when used as a value

### ISSUE THREAD

**Issue by bradzacher** [microsoft/TypeScript#57820]
### 🔎 Search Terms

cannot be used as a value because it was imported using import type

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play?ts=5.4.2#code/JYWwDg9gTgLgBDAnmApnAKnAZlCIBQA9IXAAIwDOAtCgB6oDGMNUuU+ccA5CBACZcA3PnwMIAOwrx4AXgzD8oSLDgBvJKjgBVAL7ZcBYmUo16KJizYduvAQrGT4AVzhytwoA

### 💻 Code

```ts
import type T from
// @ts-expect-error
  'mod';

const t = T;

import {type U} from
// @ts-expect-error
  'mod';

const u = U;
```

### 🙁 Actual behavior

TS does not error on the usage of `T`.

TS errors on the usage of `U` with the error

> 'U' cannot be used as a value because it was imported using 'import type'.(1361)

### 🙂 Expected behavior

TS errors on the usage of `T` with the error

> 'T' cannot be used as a value because it was imported using 'import type'.(1361)

TS errors on the usage of `U` with the error

> 'U' cannot be used as a value because it was imported using 'import type'.(1361)

**Comment by fatcerberus** [comment]
> It looks like this specifically occurs then if the imported module does not exist.

Does that matter, then?  To be clear it's still a bug since you shouldn't be able to use type imports in value positions, but in practice you already get an error on the import, and I don't think people are generally going to be `@ts-ignore`ing module resolution errors.
